com.unity.netcode.gameobjects/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -129,24 +132,59 @@ internal unsafe void __sendClientRpc(FastBufferWriter writer, uint rpcMethodId,
             };
             int messageSize;
 
+            // We check to see if we need to shortcut for the case where we are the host/server and we can send a clientRPC
+            // to ourself. Sadly we have to figure that out from the list of clientIds :( 
+            ulong serverClientId = ulong.MaxValue;
+
             if (rpcParams.Send.TargetClientIds != null)
             {
                 // Copy into a localArray because SendMessage doesn't take IEnumerable, only IReadOnlyList
                 ulong* localArray = stackalloc ulong[rpcParams.Send.TargetClientIds.Count()];
                 var index = 0;
                 foreach (var clientId in rpcParams.Send.TargetClientIds)
                 {
+                    if (clientId == NetworkManager.ServerClientId)
+                    {
+                        serverClientId = clientId;
+                        continue;
+                    }
+
                     localArray[index++] = clientId;
                 }
+
                 messageSize = NetworkManager.SendMessage(message, networkDelivery, localArray, index, true);
             }
             else if (rpcParams.Send.TargetClientIdsNativeArray != null)
             {
+                foreach (var clientId in rpcParams.Send.TargetClientIdsNativeArray)
+                {
+                    if (clientId == NetworkManager.ServerClientId)
+                    {
+                        serverClientId = clientId;
+                    }
+                }
+
                 messageSize = NetworkManager.SendMessage(message, networkDelivery, rpcParams.Send.TargetClientIdsNativeArray.Value);
```

Comment:
In the `TargetClientIds` loop above you remove the serverclientid from the list before passing to `SendMessage` but here you pass it along. I can't say that's wrong for sure, but it doesn't feel right... If it's intentional, adding a comment to the code would go a long way

Reply:
If we remove `serverCanSendToServerId` (making the behavior always the "false" behavior) then modifying the lists here won't be necessary.